scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -1682,10 +1708,23 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
    *  }}}
    */
   def caseClauses(): List[Case] = {
-    val cases = caseSeparated { caseClause() }
+    val cases = new ListBuffer[Case]
+    while (token.is[CaseIntro] || token.is[Ellipsis]) {
```

Comment:
Same reason as with `StatSep` vs `;`. `CaseIntro` is most general form of `case` that can be applied here.